### PR TITLE
Aggregated index script

### PR DESCRIPTION
**Pull request by jdelbick**
- required changes for aggregated index script
- move script to infra: https://github.com/adobe/content-lake-infra/pull/2

Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/cloud-search-index-storage.js [hunk 1]
```diff
@@ -262,11 +262,11 @@ export class CloudSearchIndexStorage {
    * @param {String} key attribute to delete by.
    * @param {String} value value of the attribute to delete by.
    */
-  async deleteBy(key, value) {
+  async deleteBy(key, value, _options) {
```
Comment (klcodanr) [inline src/cloud-search-index-storage.js:265]: Why underscrore? Should be in the JSDoc

File: src/cloud-search-index-storage.js [hunk 2]
```diff
@@ -180,11 +180,11 @@ export class CloudSearchIndexStorage {
    * @param {String} key attribute to get by.
    * @param {String} value value of the attribute to get by.
    */
-  async getObjectsBy(key, value) {
-    const searchResult = await this.#index.search(
+  async getObjectsBy(key, value, distinct = true) {
```
Comment (klcodanr) [inline src/cloud-search-index-storage.js:183]: Add to jsdoc

File: src/cloud-search-index-storage.js [hunk 3]
```diff
@@ -47,15 +47,15 @@ export class CloudSearchIndexStorage {
   ]);
   #INDEX_PREFIX = 'company';
   #log;
-  #index;
+  index;
```
Comment (klcodanr) [inline src/cloud-search-index-storage.js:50]: Why not private anymore?

**Review by klcodanr — APPROVED** [review]
Doc tweaks, but otherwise ok

File: src/cloud-search-index-storage.js [hunk 4]
```diff
@@ -47,15 +47,15 @@ export class CloudSearchIndexStorage {
...
   #indexName;
 
   constructor(context, companyId) {
     this.#indexName = context.env.ALGOLIA_CI_INDEX || `${this.#INDEX_PREFIX}-${companyId}`;
     this.#log = new ContextHelper(context).getLog();
     this.#log.info('Using Search Index', this.#indexName);
 
-    this.#index = this.getClient(context).initIndex(this.#indexName);
+    this.index = this.getClient(context).initIndex(this.#indexName);
```
Comment (jdelbick) [inline src/cloud-search-index-storage.js:58]: this is required for the scripts so we don't need to make changes to this library everytime we want to run a command that is not already supported. lmk what you think

File: src/cloud-search-index-storage.js (hunk 2 above)
Comment (jdelbick) [inline src/cloud-search-index-storage.js:183]: make distinct configurable

**Pull request merged**